### PR TITLE
fix: update expected cursor when multiple docs change

### LINKED ISSUE #1752: [Bug] `ci'`, cursor jumps to the end of line
OS: macOS 14
nvim: 0.9.5
vscode-nvim: 1.5.0

vscode-neovim behavior (cursor jumps to the end of the line):

https://github.com/vscode-neovim/vscode-neovim/assets/28700378/238e82cd-8bfc-4e7b-a25d-1cc9b2758bde

Neovim behavior (cursor stays inside the quotes):

https://github.com/vscode-neovim/vscode-neovim/assets/28700378/b71d0191-ddeb-4c1e-8063-e4bf10b39f83

Code:

```styl
$base = '../rewards'
$bgi = $proxy($base, 'bgi')

@import '../includes/highlight.styl'

.open-task-btn
  $bgi '../static/rewards/open-tasks-btn-double.png'
  $highlight '../static/rewards/open-tasks-btn-double.png'
```

### PR DESCRIPTION
**Pull request by xiyaowong**
fixes #1752

### CODE REVIEW

**Comment by ollien** [comment]
Is this meant to just be a test?

File: src/cursor_manager.ts [hunk 1]
```diff
@@ -61,8 +61,17 @@ export class CursorManager implements Disposable {
      * cursor updates while entering insert mode and insert mode commands. Thus, when those events occur,
      * this flag is used to disable ignoring the update. This is set to true when entering insert
      * mode or running insert mode command, and set to false before document updates in insert mode.
+     *
+     * The flag corresponds to each `TextEditor`
      */
-    public wantInsertCursorUpdate = true;
+    private _wantInsertCursorUpdate: WeakMap<TextEditor, boolean> = new WeakMap();
+    public wantInsertCursorUpdate = (editor: TextEditor) => this._wantInsertCursorUpdate.get(editor) ?? false;
```
Comment (ollien) [inline src/cursor_manager.ts:68]: nit: is there any reason for these not to be methods?

**Comment by ollien** [comment]
I think I understand the changes, but why was this affecting only macOS? Could you shed some light on your changes?

**Comment by xiyaowong** [comment]
I can reproduce on Windows

**Comment by xiyaowong** [comment]
You can see the failed tests, they failed on all systems.

**Comment by ollien** [comment]
Interesting, given the activity on the original report. Huh

**Review by ollien — APPROVED** [review]

**Review by theol0403 — APPROVED** [review]

File: src/cursor_manager.ts (hunk 1 above)
Comment (xiyaowong) [reply src/cursor_manager.ts:68]: Personal preference, looks more compact. 😹

**Pull request merged**